pkg/controller/subscription/referredconfigmap_handler.go

```diff
@@ -0,0 +1,215 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscription
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appv1alpha1 "github.com/IBM/multicloud-operators-subscription/pkg/apis/app/v1alpha1"
+	"github.com/IBM/multicloud-operators-subscription/pkg/utils"
+)
+
+//SercertReferredMarker is used as a label key to filter out the secert coming from reference
+var SercertReferredMarker = "IsReferredBySub"
+
+func (r *ReconcileSubscription) ListAndDeployReferredConfigMap(refCfg *corev1.ConfigMap, instance *appv1alpha1.Subscription) error {
+	if klog.V(utils.QuiteLogLel) {
+		fnName := utils.GetFnName()
+		klog.Infof("Entering: %v()", fnName)
+
+		defer klog.Infof("Exiting: %v()", fnName)
+	}
+
+	// list secret within the sub ns given the secert name
+	localCfg, err := r.ListReferredConfigMapByName(instance, refCfg)
+
+	if err != nil {
+		return err
+	}
+
+	// list the used secert marked with the subscription
+	instanceKey := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
+	oldCfgs, err := r.ListReferredConfigMap(instanceKey)
+
+	if err != nil {
+		return err
+	}
+	// if the listed secert is used by myself, and it's not the newOne, then delete it, otherwise,
+	err = r.UpdateLabelsOnOldRefConfigMap(instance, refCfg, oldCfgs)
+
+	if err != nil {
+		return err
+	}
+
+	if localCfg == nil {
+		//delete old lalbels
+		err = r.DeployReferredConfigMap(instance, refCfg)
+		return err
+	}
+
+	// we already have the referred secert in the subscription namespace
+	if localCfg.GetName() != "" {
+		lb := localCfg.GetLabels()
+		lb[instance.GetName()] = "true"
+		lb[SercertReferredMarker] = "true"
+		localCfg.SetLabels(lb)
+		err = r.Client.Update(context.TODO(), localCfg)
+	}
+
+	return err
+}
+
+//ListReferredSecret lists secert within the subscription namespace and having label <subscription.name>:"true"
+func (r *ReconcileSubscription) ListReferredConfigMapByName(instance *appv1alpha1.Subscription, ref *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+	refKey := types.NamespacedName{Namespace: instance.GetNamespace(), Name: ref.GetName()}
+	localRef := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), refKey, localRef)
+
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return localRef, nil
+}
+
+func getLabelOfSubscription(subName string) *metav1.LabelSelector {
```

Comment:
if it is internal func and is not member of a struct, let's put it into utils. 

I don't think the name matches what the function does here.